### PR TITLE
Fills inputs for task

### PR DESCRIPTION
**Pull request by olegshtch**
Updates minimum gradle version due incompatible changes in TaskInputs methods.

### CODE REVIEW

**Comment by vojtechhabarta** [comment]
@olegshtch thanks for this PR.

Can anybody comment/review this PR?
I think updating gradle to version 3.0 is fine now.

File: typescript-generator-gradle-plugin/src/main/java/cz/habarta/typescript/generator/gradle/TypeScriptGeneratorPlugin.java [hunk 1]
```diff
@@ -16,6 +16,7 @@ public void apply(Project project) {
         for (Task task : project.getTasks()) {
```
Comment (rbrindl) [inline typescript-generator-gradle-plugin/src/main/java/cz/habarta/typescript/generator/gradle/TypeScriptGeneratorPlugin.java]: To facilitate configuration avoidance and lazy task initialization:
```
project.getTasks().matching{ it.getName().startsWith('compile') }.all { Task task ->
    generateTsTask.getInputs().files(task)
}
```
Adding the task directly as input implicitly creates a task dependency and an explicit dependsOn is not needed.
The `matching.all` construct does not materialize the tasks and task initialization is deferred.

Comment (olegshtch) [reply typescript-generator-gradle-plugin/src/main/java/cz/habarta/typescript/generator/gradle/TypeScriptGeneratorPlugin.java]: @rbrindl It's still require to upgrade gradle due different signatures: https://docs.gradle.org/2.1/javadoc/org/gradle/api/tasks/TaskInputs.html#files(java.lang.Object...) and
https://docs.gradle.org/3.0/javadoc/org/gradle/api/tasks/TaskInputs.html#files(java.lang.Object...)
Without it I get error:
`Caused by: java.lang.NoSuchMethodError: org.gradle.api.tasks.TaskInputs.files([Ljava/lang/Object;)Lorg/gradle/api/tasks/TaskInputs;`

**Comment by olegshtch** [comment]
Rebased. Use task itself without resolving files.

**Comment by olegshtch** [comment]
Is something left to do in this PR?

**Comment by vojtechhabarta** [comment]
I am not sure. Can anybody comment/review it? (@rbrindl)

**Pull request merged**